hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3037,18 +3037,51 @@ func deleteNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace)
 }
 
 func deleteControlPlaneOperatorRBAC(ctx context.Context, c client.Client, rbacNamespace string, controlPlaneNamespace string) error {
-	rbacMeta := metav1.ObjectMeta{
-		Name:      "control-plane-operator-" + controlPlaneNamespace,
-		Namespace: rbacNamespace,
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "control-plane-operator-" + controlPlaneNamespace,
+			Namespace: rbacNamespace,
+		},
 	}
-
-	err := c.Delete(ctx, &rbacv1.Role{ObjectMeta: rbacMeta})
-	if err != nil && !apierrors.IsNotFound(err) {
+	err := c.Get(ctx, client.ObjectKeyFromObject(role), role)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
This looks like a bug here and in the following similar operations — if the resource isn't found, shouldn't you proceed to try deleting other things rather than returning early?

Reply:
gah yes.  nice find!  two resources being deleted in the same function threw me off.